### ISSUE TITLE
False positive for RCS1074 depending on property ordering

### ISSUE THREAD

**Issue by cbersch** [JosefPihrt/Roslynator#1767]
The following example raises a wrong RCS1074 diagnostic in L7 (tested with Roslynator 4.15.0, .NET 10.0.300)
```csharp
public readonly struct Foo
{
    public bool B { get; init; }

    public bool A { get; init; } = true;

    public Foo()
    {
    }
}
```

<img width="314" height="238" alt="Image" src="https://github.com/user-attachments/assets/db3055b7-a160-4c4d-a5dc-439d9eba2e42" />

Wrong, because removing the constructor leads to error [CS8983](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/constructor-errors?f1url=%3FappId%3Droslyn%26k%3Dk(CS8983)#constructors-in-struct-types): A 'struct' with field initializers must include an explicitly declared constructor.

After changing the order of properties `A` and `B` the diagnostic is not reported:

```csharp
public readonly struct Foo
{
    public bool A { get; init; } = true;

    public bool B { get; init; }

    public Foo()
    {
    }
}
```
